test/e2e/create_cluster_test.go

```diff
@@ -21,7 +22,7 @@ func TestNoneCreateCluster(t *testing.T) {
 	clusterOpts := globalOpts.DefaultClusterOptions()
 	clusterOpts.ControlPlaneAvailabilityPolicy = "SingleReplica"
 
```

Comment:
should hyperv1.AWSPlatform be set dynamically and we have a single TestCreateCluster which we always run?

Reply:
Basically same thoughts as https://github.com/openshift/hypershift/pull/884#discussion_r788889185.

Hopefully with this step it should at least be more clear at the moment how platform behaviors diverge (as all the platform specific logic is now clearly in one place, `CreateCluster()`).

So just a step forward, with some bonus bugfixes. Further design is required after a more holistic analysis incorporating the CLI code (which also suffers from these issues and is growing organically).